wgengine/wgcfg/config.go

```diff
@@ -31,10 +29,59 @@ type Config struct {
 type Peer struct {
 	PublicKey           wgkey.Key
 	AllowedIPs          []netaddr.IPPrefix
-	Endpoints           string // comma-separated host/port pairs: "1.2.3.4:56,[::]:80"
+	Endpoints           Endpoints
 	PersistentKeepalive uint16
 }
 
+// Endpoints represents the routes to reach a remote node.
+// It is serialized and provided to wireguard-go as a conn.Endpoint.
+type Endpoints struct {
+	// PublicKey is the public key for the remote node.
+	PublicKey wgkey.Key `json:"pk"`
+	// DiscoKey is the disco key associated with the remote node.
+	DiscoKey tailcfg.DiscoKey `json:"dk,omitempty"`
+	// IPPorts is a set of possible ip+ports the remote node can be reached at.
+	IPPorts []netaddr.IPPort `json:"ipp,omitempty"`
```

Comment:
slice in a value type is a little scary. normally the rule is no value types with mutable pointers.

can we use some immutable IPPortSet type instead? Or flatten it to a comma-separated strings with a method to get them back out parsed?